Make Keycloak Model Map JPA module JDK 17 ready

### Description

I built Keycloak using OpenJDK 17.0.2 and maven 3.8.4 (using `mvn clean install -DskipTests -Pdistribution` on latest main). The build failed for module `Keycloak Model Map JPA` with the following error:

```
[ERROR] Failed to execute goal org.hibernate.orm.tooling:hibernate-enhance-maven-plugin:5.3.20.Final:enhance (default) on project keycloak-model-map-jpa: Unable to enhance class: JpaRoleMetadata.class: Failed to enhance class org.keycloak.models.map.storage.jpa.role.entity.JpaRoleMetadata: Unsupported class file major version 61 -> [Help 1]
```
So atm a (not currently planned) update to use JDK17 as default would fail due to the Problem with the enhancer. So the enhancer needs to be updated i guess.


### Discussion

_No response_

### Motivation

Have an easy mid-term upgrade path to use more modern java versions.

### Details

_No response_

--- a/pom.xml
+++ b/pom.xml
@@ -82,8 +82,8 @@
         <dom4j.version>2.1.3</dom4j.version>
         <h2.version>1.4.197</h2.version>
         <jakarta.persistence.version>2.2.3</jakarta.persistence.version>
-        <hibernate.core.version>5.3.20.Final</hibernate.core.version>
-        <hibernate.c3p0.version>5.3.20.Final</hibernate.c3p0.version>
+        <hibernate.core.version>5.3.24.Final</hibernate.core.version>
+        <hibernate.c3p0.version>5.3.24.Final</hibernate.c3p0.version>
         <infinispan.version>12.1.7.Final</infinispan.version>
         <infinispan.protostream.processor.version>4.4.1.Final</infinispan.protostream.processor.version>
         <javax.annotation-api.version>1.3.2</javax.annotation-api.version>

